pkg/cloud/services/compute/instance.go

```diff
@@ -504,6 +505,25 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 		return nil, err
 	}
 
+	if portOpts.PortSecurity != nil {
```

Comment:
seems CreateOpts has those 2 definition
```
SecurityGroups      *[]string     `json:"security_groups,omitempty"`
        AllowedAddressPairs []AddressPair `json:"allowed_address_pairs,omitempty"`
```

why we don't combine this with create function ?

Reply:
I will have to test that. My worry is that OpenStack would apply the security group to the node if its set on the instance regardless of the settings on the port.

Reply:
If I am not mistaken, in my tests when security groups are applied to an instance they can overwrite the settings on the port. I can double check this though, it was a long time ago.